honeybee_3dm/togeometry.py

```diff
@@ -1,34 +1,96 @@
+        to describe how curved faces should be converted into planar elements.
+        If None, Rhino3dm's Default Meshing Parameters will be used.
+    """
 
+    # This is a container
+    faces = []
 
-def to_face3d(geo):
-    """List of Ladybug Face3D objects from a Rhino Brep, Surface or Mesh."""
-    pass
+    # If it's a Mesh
+    if isinstance(geo, rhino3dm.Mesh):
+        mesh = geo
+        # Get all the vertices
+        pts = [to_point3d(mesh.Vertices[i]) for i in range(len(mesh.Vertices))]
+        for j in range(len(mesh.Faces)):
+            face = mesh.Faces[j]
+            if len(face) == 4:
+                all_verts = (pts[face[0]], pts[face[1]],
+                             pts[face[2]], pts[face[3]])
+            else:
+                all_verts = (pts[face[0]], pts[face[1]], pts[face[2]])
+            # Create Ladybug Face3D objects based on tuples of vertices
+            faces.append(Face3D(all_verts))
 
+    # If it's an Extrusion
+    if isinstance(geo, rhino3dm.Extrusion):
+        # Convert it into a Mesh first
+        mesh = geo.GetMesh(rhino3dm.MeshType.Default)
+        if isinstance(mesh, rhino3dm.Mesh):
+            # Get all the vertices
+            pts = [to_point3d(mesh.Vertices[i])
+                   for i in range(len(mesh.Vertices))]
+            for j in range(len(mesh.Faces)):
+                face = mesh.Faces[j]
+                if len(face) == 4:
+                    all_verts = (pts[face[0]], pts[face[1]],
+                                 pts[face[2]], pts[face[3]])
+                else:
+                    all_verts = (pts[face[0]], pts[face[1]], pts[face[2]])
+                # Create Ladybug Face3D objects based on tuples of vertices
+                faces.append(Face3D(all_verts))
 
-def to_polyface3d(geo):
-    """A Ladybug Polyface3D object from a Rhino Brep."""
-    pass
+    # If it's a Brep
+    if isinstance(geo, rhino3dm.Brep):
+        # Convert it into a list of Meshes
+        meshes = [geo.Faces[f].GetMesh(rhino3dm.MeshType.Any) for f in range(
+            len(geo.Faces)) if type(geo.Faces[f]) != list]
 
+        # For each Mesh in the list create a Ladybug Face3D object and add to
+        # a container
+        for i in range(len(meshes)):
+            if isinstance(meshes[i], rhino3dm.Mesh):
+                mesh = meshes[i]
+                # Get all the vertices
+                pts = [to_point3d(mesh.Vertices[i])
+                       for i in range(len(mesh.Vertices))]
+                for j in range(len(mesh.Faces)):
+                    face = mesh.Faces[j]
+                    if len(face) == 4:
+                        all_verts = (pts[face[0]], pts[face[1]],
+                                     pts[face[2]], pts[face[3]])
+                    else:
+                        all_verts = (pts[face[0]], pts[face[1]], pts[face[2]])
+                    # Create Ladybug Face3D objects based on tuples of vertices
+                    faces.append(Face3D(all_verts))
 
```

Comment:
You need a final check here for unsupported objects which are not Mesh, Extrusion or Brep to give an error or a warning to use so they know those object types are not supported.